internal/server/data/group.go

```diff
@@ -19,12 +19,19 @@ func ListGroups(db *gorm.DB, selectors ...SelectorFunc) ([]models.Group, error)
 	return list[models.Group](db, selectors...)
 }
 
-func ListIdentityGroups(db *gorm.DB, identityID uid.ID) (result []models.Group, err error) {
-	if err := db.Model(&models.Identity{Model: models.Model{ID: identityID}}).Association("Groups").Find(&result); err != nil {
-		return nil, err
+func WhereGroupIncludesUser(id uid.ID) SelectorFunc {
+	return func(db *gorm.DB) *gorm.DB {
+		return db.
+			Joins("JOIN identities_groups ON groups.id = identities_groups.group_id").
+			Where("identities_groups.identity_id = ?", id)
 	}
+}
 
-	return result, nil
+// TODO: replace calls with ListGroups(..., WhereGroupIncludesUser(id))
+func ListIdentityGroups(db *gorm.DB, identityID uid.ID) ([]models.Group, error) {
```

Comment:
This query is used from other endpoints. It seems like it is equivalent to `ListGroups` with `WhereGroupIncludesUser`. I'm thinking we only want one way to do this.

Reply:
would just go ahead and remove this.